### PR TITLE
Added auto instrumentation scripts and updated GH workflow for release

### PR DESCRIPTION
**Pull request by AsakerMohd**
*Issue #, if available:*

*Description of changes:*
Updated the installation script templates so that when we build the application, the scripts are built accordingly. Also updated the templates to use the AWS Otel Distro instead of the upstream Otel Distro. I also updated the README to reflect that we are using DockerFiles for windows images that fall under the [Windows EULA Agreement](https://learn.microsoft.com/en-us/virtualization/windowscontainers/images-eula)

*Testing:*
Tested that the scripts on both linux and windows and made sure they are working as expected. Set the version to 1.2.0 since that's the latest release and ran the script aws-otel-dotnet-install.sh which basically downloads the 1.2.0 from the releases page and unpacks the zip file. Replaced the unpacked contents with the latest contents by running build.sh and copying the files from the OpentelemetryDistro folder. Finally ran the instrument.sh file to add the instrumentation env vars and when starting the sample app, it was instrumented as expected. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/post_release_version_bump.yml [hunk 1]
```diff
@@ -98,8 +98,10 @@ jobs:
         run: |
           DEV_VERSION="${{ github.event.inputs.version }}.dev0"
           sed -i "s/public static string version = \".*\";/public static string version = \"${DEV_VERSION}\";/" src/AWS.Distro.OpenTelemetry.AutoInstrumentation/Version.cs
+          sed -i "s/public static string version = \".*\";/public static string version = \"${VERSION}\";/" build/Version.cs
```
Comment (vastin) [inline .github/workflows/post_release_version_bump.yml:101]: Should it use ${DEV_VERSION} here?

File: build/Build.InstallationScripts.cs [hunk 2]
```diff
@@ -9,6 +9,8 @@ internal partial class Build : NukeBuild
 {
     private readonly AbsolutePath installationScriptsFolder = RootDirectory / "bin" / "InstallationScripts";
 
+    public static string version = "1.2.0.dev0";
```
Comment (vastin) [inline build/Build.InstallationScripts.cs:12]: Please address the warnings.

Comment (AsakerMohd) [reply build/Build.InstallationScripts.cs:12]: let me check those.

File: .github/workflows/post_release_version_bump.yml (hunk 1 above)
Comment (AsakerMohd) [reply .github/workflows/post_release_version_bump.yml:101]: yes good catch. Let me update that.

**Review by vastin — APPROVED** [review]

**Pull request merged**